internal/stores/gcs.go

```diff
@@ -106,10 +96,16 @@ func (s *GCSStore) Upload(ctx context.Context, objects ...string) error {
 		// Reset to the start of the file because metadata generation has already read it once
 		_, err = f.Seek(0, io.SeekStart)
 		if err != nil {
+			if err := cleanup(); err != nil {
```

Comment:
If cleanup fails here, then we have no idea why it failed (or what the original error was). 

Maybe we can have cleanup take the err, wrap it and return a combined error from f.Seek if cleanup fails, otherwise just return the original err.

Reply:
I added more error context and set up multierror. Let me know if this makes it better or worse 😅

Reply:
yep good.